src/Pages/MyAccount/Components/PaymentMethod/RegistCardPage.js

```diff
@@ -0,0 +1,94 @@
+import React, { Component } from "react";
+import styled from "styled-components";
+
+class RegistCardPage extends Component {
+  constructor() {
+    super();
+    this.state = {
+      cardNumber: "",
+      confirmCCN: true,
+      expirydate: "",
+      confirmdate: true,
+    };
+  }
+
+  CardNumberHandler = (e) => {
+    if (e.target.value.length > 19) {
+    } else if (e.target.value.length !== 0) {
+      let ccn = e.target.value;
+      ccn = ccn.split(" ").join("");
+      ccn = ccn.match(new RegExp(".{1,4}$|.{1,4}", "g")).join(" ");
+      console.log(ccn);
+      this.setState({ cardNumber: ccn });
+    } else {
+      this.setState({ cardNumber: e.target.value });
+    }
+
+    if (e.target.value.length === 19) {
+      let temp = e.target.value.replaceAll(" ", "");
+      console.log(temp);
+      let result = 0;
+      for (let i = 0; i < temp.length; i++) {
+        let count = (i % 2) + 1;
+        let resulttemp = Number(temp[i]) * count;
+        resulttemp = resulttemp.toString();
+        if (resulttemp.length === 1) {
+          resulttemp = "0" + resulttemp;
+        }
+        result = result + Number(resulttemp[0]) + Number(resulttemp[1]);
+      }
+      if (result === 55) {
+        this.setState({ confirmCCN: true });
+      } else {
+        this.setState({ confirmCCN: false });
+      }
+    }
+  };
+
+  ExpirydateHandler = (e) => {
+    let str = e.target.value;
+    if (str.length > 5) {
+      console.log("you can't more input date");
+    } else if (str.length !== 0) {
+      let date = str;
+      date = date.split("/").join("");
+      date = date.match(new RegExp(".{1,2}$|.{1,2}", "g")).join("/");
+      console.log(date);
+      this.setState({ expirydate: date });
+    } else {
+      this.setState({ expirydate: str });
+    }
```

Comment:
```suggestion
    const date = e.target.value;
    const isValid = date.length !== 0
    if (date.length > 5) return alert("you can't more input date");
    
    const dateParse = date.split("/").join("").match(new RegExp(".{1,2}$|.{1,2}", "g")).join("/");
    this.setState({ expiryDate: isValid ? dateParse : date });
  }
```
- 변수는 가급적 const로 선언해주세요
- console.log가 아니라 alert로 띄워야할 내용 같습니다
- 카멜케이스~
- setState는 가급적 한번만 사용하기~

Reply:
아, 죄송합니다. 경고문은 카드창에 포함된 div박스에 리폼해서 보여줄예정인데 미구현상태라서 콘솔로만 구현한 상태입니다. 그외 사항에대해서는 잘 참고해서 바꿔보겠습니다. 감사합니다.